tests/integration/test_kyber.py

```diff
@@ -0,0 +1,139 @@
+# Prefilled account from Kyber Ganache snapshot
+WALLET_PRIVATE_KEY = 0x979d8b20000da5832fc99c547393fdfa5eef980c77bfb1decb17c59738d99471
+
+KNC_TOKEN = Erc20Token(
+    ticker="KNC",
+    wei_ticker="KEI",
+    addresses={"ganache": kyber_snapshot_addresses.TokenAddress.KNC.value},
+    network="ganache"
+)
+
+
+def take_snapshot():
+    body = dict(jsonrpc="2.0", method="evm_snapshot", params=[], id=1)
+    response = requests.post(f"http://localhost:{PORT}", json=body)
+    return response.json()["result"]
+
+
+def revert_to_snapshot(snapshot_id):
+    body = dict(jsonrpc="2.0", method="evm_revert", params=[snapshot_id], id=1)
+    requests.post(f"http://localhost:{PORT}", json=body)
+
+
+@pytest.fixture
+def test_account(monkeypatch):
+    monkeypatch.setattr(Account, "generate_private_key", lambda: to_bytes(WALLET_PRIVATE_KEY))
+    account = Account.create(TESTING_KEYSTORE_FOLDER, "test_password")
+    yield account
+    account.keystore_file_path.unlink()
+
+
+@pytest.fixture()
+def patch_kyber_support(monkeypatch):
+    monkeypatch.setitem(NETWORK_ADDRESS_MODULES_BY_CHAIN_ID, CHAIN_ID, kyber_snapshot_addresses)
+    monkeypatch.setattr(Kyber, "SUPPORTED_NETWORKS", ["ganache"])
+
+
+@pytest.fixture()
+def patch_network(monkeypatch):
+    class Ganache(Network):
+        pass
+
+    monkeypatch.setitem(NETWORK_CLASSES, "ganache", Ganache)
+    monkeypatch.setitem(Network.CHAIN_ID_MAPPING, "ganache", CHAIN_ID)
+
+
+@pytest.fixture(scope="module")
+def kyber_chain():
+    proc = psutil.Popen(
+        GANACHE_COMMAND,
+        cwd=str(FAKE_BLOCKCHAIN_PATH)
+    )
+    time.sleep(2)
+    yield
+    proc.terminate()
+    proc.wait()
+
+
+@pytest.fixture()
+def kyber(test_account: Account, kyber_chain, patch_network):
+    snapshot_id = take_snapshot()
+    w3 = make_web3_provider(f"http://localhost:{PORT}", test_account)
+    yield Kyber(w3)
+    revert_to_snapshot(snapshot_id)
+
+
+def test_buy_tokens(test_account, patch_kyber_support, kyber):
+    knc_balance_before = get_token_balance(kyber.w3, test_account, KNC_TOKEN)
+    buy_amount = TokenAmount(10, KNC_TOKEN)
+    kyber.buy_tokens(test_account, buy_amount)
+    knc_balance_after = get_token_balance(kyber.w3, test_account, KNC_TOKEN)
+    assert knc_balance_after == knc_balance_before + buy_amount
+
+
+def test_cannot_buy_on_unsupported_network(test_account, kyber):
+    with pytest.raises(ExchangeError):
+        kyber.buy_tokens(test_account, TokenAmount(10, KNC_TOKEN))
+
+
+def test_cannot_buy_zero_tokens(test_account, patch_kyber_support, kyber):
+    with pytest.raises(ExchangeError):
+        kyber.buy_tokens(test_account, TokenAmount(0, KNC_TOKEN))
+
+
+def test_cannot_buy_without_eth(test_account, patch_kyber_support, kyber):
+    tx = {
+        "to": "0x0000000000000000000000000000000000000000",
+        "from": test_account.address,
+        "value": test_account.get_ethereum_balance(kyber.w3).as_wei,
+        "gasPrice": 0,
+    }
+    kyber.w3.eth.sendTransaction(tx)
```

Comment:
I would add
```suggestion
    kyber.w3.eth.sendTransaction(tx)
    assert test_account.get_ethereum_balance(kyber.w3).as_wei == 0, "account not empty"
```
to make sure the account is actually drained before proceeding with the test.

Reply:
Wouldn't it make more sense to wait for the transaction to get included in a block?